server/memphis_handlers_user_mgmt.go

```diff
@@ -402,6 +434,131 @@ func (umh UserMgmtHandler) AuthenticateNatsUser(c *gin.Context) {
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	var message string
+	if exist {
+		message = "Only root user exists"
+	} else {
+		message = "More than root user exists"
+	}
+	serv.Warnf(message)
+	c.IndentedJSON(200, gin.H{"exist": exist, "message": message})
+}
+
+func (umh UserMgmtHandler) AddUserSignUp(c *gin.Context) {
+	exist, err := isOnlyRootUserExist()
+	if err != nil {
+		serv.Errorf("AddUserSignUp error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	if !exist {
+		serv.Warnf("More than root user exists")
+		c.IndentedJSON(401, gin.H{"message": "More than root user exists"})
+		return
+	} else {
+		serv.Warnf("Only root user exists")
+		var body models.AddUserSchema
+		ok := utils.Validate(c, &body, false, nil)
+		if !ok {
+			return
+		}
+		username := strings.ToLower(body.Username)
+		exist, _, err := IsUserExist(username)
+		if err != nil {
+			serv.Errorf("AddUserSignUp error: " + err.Error())
+			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+			return
+		}
+		if exist {
+			serv.Warnf("A user with this username is already exist")
+			c.AbortWithStatusJSON(configuration.SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": "A user with this username is already exist"})
+			return
+		}
+
+		usernameError := validateUsername(username)
+		if usernameError != nil {
+			serv.Warnf(usernameError.Error())
+			c.AbortWithStatusJSON(configuration.SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": usernameError.Error()})
+			return
+		}
+		email := strings.ToLower((body.Email))
+		fullName := strings.ToLower(body.FullName)
+		password := strings.ToLower(body.Password)
+		subscription := body.Subscribtion
+
+		emailError := validateEmail(email)
+		if emailError != nil {
+			serv.Warnf(emailError.Error())
+			c.AbortWithStatusJSON(configuration.SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": emailError.Error()})
+			return
+		}
+
+		if subscription {
+			mailchimpClient := gochimp3.New(configuration.MAILCHIMP_KEY)
+			mailchimpListID := configuration.MAILCHIMP_LIST_ID
+			mailchimpList, err := mailchimpClient.GetList(mailchimpListID, nil)
+			if err != nil {
+				serv.Errorf("SignUperror: " + err.Error())
+			}
+
+			mailchimpReq := &gochimp3.MemberRequest{
+				EmailAddress: email,
+				Status:       "subscribed",
+				Tags:         []string{"signup"},
+			}
+
+			if _, err := mailchimpList.CreateMember(mailchimpReq); err != nil {
+				serv.Errorf("Failed to subscribe in mailChimp: " + err.Error())
+			}
+		}
+
+		newUser := models.User{
+			ID:              primitive.NewObjectID(),
+			Username:        username,
+			Password:        password,
+			FullName:        fullName,
+			Email:           email,
+			Subscribtion:    subscription,
+			UserType:        "managment",
+			CreationDate:    time.Now(),
+			AlreadyLoggedIn: false,
+			AvatarId:        1,
+		}
+
+		_, err = usersCollection.InsertOne(context.TODO(), newUser)
+		if err != nil || len(username) == 0 {
```

Comment:
why are you checking here the length of the username, it should happn inside the validateUsername function